File: javascript/leetcode/single-number-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    seen = {}
    for (var i = 0; i < nums.length; i++) {
        seen[nums[i]] = seen[nums[i]] ? seen[nums[i]] + 1 : 1
    }
    for (k in seen) {
        if (seen[k] == 1) { return parseInt(k) }
    }
};

File: ruby/leetcode/single-number-ii.rb
# @param {Integer[]} nums
# @return {Integer}
def single_number(nums)
    seen = {}
    nums.each do |i|
        seen[i] = if seen[i] then seen[i] + 1 else 1 end
    end
    seen.each do |k, v|
        if v == 1
            return k
        end
    end
end
